pandas/core/categorical.py

```diff
@@ -263,18 +259,12 @@ def __init__(self, values, categories=None, ordered=None, name=None, fastpath=Fa
 
         if categories is None:
             try:
-                codes, categories = factorize(values, sort=True)
-                # If the underlying data structure was sortable, and the user doesn't want to
-                # "forget" this order, the categorical also is sorted/ordered
-                if ordered is None:
-                    ordered = True
+                codes, categories = factorize(values, sort=ordered)
```

Comment:
This was the disccusion in #9346 and especially #9347. This would make the categories sorted by order of appearance if `ordered==False` (which it is now per default). Before the categories were almost always lexi-ordered.